src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -759,24 +759,28 @@ def cleanup_nodes(self) -> bool:
         to_reimage = []
         to_delete = []
 
-        nodes = Node.search_states(
-            scaleset_id=self.scaleset_id, states=NodeState.ready_for_reset()
-        )
-
-        if not nodes:
-            logging.info("no nodes need updating: %s", self.scaleset_id)
-            return False
-
         # ground truth of existing nodes
         azure_nodes = list_instance_ids(self.scaleset_id)
 
+        nodes = Node.search_states(scaleset_id=self.scaleset_id)
```

Comment:
I expect a significantly small portion the Nodes to be in the ready_for_reset states at any given time.  With early exiting based on this expectation (previously on line 768), it made sense to not retrieve all of the nodes up front.

This PR removes the early exit, which is fine.  However, as a byproduct, we should not perform two queries against Node.

Instead, this PR should do something like this:

```python
nodes = Node.search_states(scaleset_id=self.scaleset_id)
nodes_to_reset = [x for x in nodes if x.state in NodeState.ready_for_reset()]
```